src/components/ContactsList/index.js

```diff
@@ -0,0 +1,145 @@
+/* global cozy */
+import React, { Component } from "react";
+import ContactsError from "./ContactsError";
+
+let sampleId = 0;
+const sampleContacts = [
+  {
+    _id: sampleId++,
+    name: { givenName: "Amélie", familyName: "Poulain" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Edward", familyName: "Snowden" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Aylin", familyName: "" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Benjamin", familyName: "André" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Bertrand", familyName: "Alevin" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "André", familyName: "Manoukian" },
+    avatar: ""
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Erika", familyName: "Shell" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Brendan", familyName: "Abolivier" },
+    avatar: "",
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "COZY", familyName: "" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "EDF", familyName: "" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "CAF", familyName: "" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  },
+  {
+    _id: sampleId++,
+    name: { givenName: "Eric", familyName: "D" },
+    avatar: "",
+    phone: [{ number: "06 12345678", primary: true }],
+    email: [{ address: "jeanneige@mail.com", primary: true }]
+  }
+];
+
+export const contactSort = (contact, comparedContact) => {
+  const firstNameA =
+    contact.name && contact.name.givenName
+      ? contact.name.givenName.toUpperCase()
+      : "";
+  const firstNameB = comparedContact.name
```

Comment:
This should be `const firstNameB = comparedContact. name && comparedContact. givenName `.
(I ran into this error, or I would never have spotted it!)